README.md

```diff
@@ -333,25 +333,106 @@ _You can enable the following settings in Xcode by running [this script](resourc
+
+  <details>
+
   ```swift
-  enum Direction {
-    case left
-    case right
-  }
+  // WRONG
+  let sun: Star = .init(mass: 1.989e30)
+  let earth: Planet = .earth
 
-  func someDirection() -> Direction {
-    // WRONG
-    return Direction.left
+  // RIGHT
+  let sun = Star(mass: 1.989e30)
+  let earth = Planet.earth
 
-    // RIGHT
-    return .left
+  // ALSO RIGHT: Explicit types are required when there is no right-hand-side value.
+  let sun: Star
+  let earth: Planet
+
+  // ALSO RIGHT: Explicit types can be necessary when the right-hand side has
+  // a different type from the one written explicitly on the left-hand side.
+  let naturalSatellite: PlanetaryBody? = Moon(mass: 7.347e22)
+  let moon: PlanetaryBody? = nil // nil literals are typeless
+  let numberOfPlanets: UInt = 8 // integer literals default to `Int`
+  let sunMass: CGFloat = 1.989e30 // floating-point literals default to `Double`
+  let planets: [Planet] = [] // empty collection literals are typeless
+
+  // ALSO RIGHT: Some of the examples above can also be written idiomatically without an explicit type on
+  // the left-hand-side, by instead giving the right-hand side value an explicit type. Either style is fine.
+  let numberOfPlanets = UInt(8)
+  let sunMass = CGFloat(1.989e30)
+  let planets = [Planet]()
+  ```
+  
+  There are some rarer cases where the inferred type syntax has a different meaning than the explicit type syntax. In these cases, the explicit type syntax is still permitted:
+  
+  ```swift
+  extension String {
+    static let earth = "Earth"
+  }
+
+  // RIGHT: If the property's type is optional, moving the optional type 
+  // to the right-hand side may result in invalid code.  
+  let planetName: String? = .earth
+  
+  // WRONG: fails with "error: type 'String?' has no member 'foo'"
+  let planetName = String?.earth
+  ```
+  
+  ```swift
+  struct SaturnOutline: ShapeStyle { ... }
+  
+  extension ShapeStyle where Self == SaturnOutline {
+    static var saturnOutline: SaturnOutline { 
+      SaturnOutline() 
+    }
   }
+  
+  // RIGHT: If the property's type is an existential / protocol type, moving the type
+  // to the right-hand side will result in invalid code if the value is defined in an
+  // extension like `extension ShapeStyle where Self == SaturnOutline`.
+  // SwiftFormat autocorrect detects this case by checking for the existential `any` keyword.
+  let myShape1: any ShapeStyle = .saturnOutline
+  
+  // WRONG: fails with "error: static member 'saturnOutline' cannot be used on protocol metatype '(any ShapeStyle).Type'"
+  let myShape2 = (any ShapeStyle).myShape
```

Comment:
By the way -- for this we have the ability to tell SwiftFormat a list of symbols to exclude from this rule (`--preservesymbols ShapeStyle,OtherType,etc). Inside the apps repo we could exclude any type or property name where we frequently use this syntax. Two relevant examples are:
 - `UnresolvedShapeStyle` 
   - all existing callsites use `any UnresolvedShapeStyle` so this is already handled properly
 - `canonicalFake`
   - there are about a dozen examples of `let experimentsService: ExperimentsService = .canonicalFake`. It seems like a good idea to exclude this using `--preservesymbols canonicalFake`. Updating the callsites to `let experimentsService: any ExperimentsService = .canonicalFake` also works.

Reply:
Thanks for flagging this @calda . I was wondering about this... My recollection is that eventually it's going to be required that we use `any` for any existential/protocol types but that this requirement is being slowly rolled out. Is your understanding also that eventually we will always need to prefix an existential/protocol type with `any`? If so, I think it's less problematic if we need to work around certain situations like the above ones you highlighted for the time being.

Reply:
I am trying to understand if there would be cases when we still need `--preservesymbols` if/when the language requires that all existential/protocol types are preceded by `any`.

Reply:
I saw on the Swift Evolution forums that they decided to not make `any` mandatory in Swift 6 mode after all. Although after a quick search on the forums I can't actually find where they said that. 

This doesn't necessarily confirm it, but if you build code like [this](https://swiftfiddle.com/jazi4p4ji5eftlgxzvigkq2wi4) using a nightly Swift 6 build it doesn't require `any`.

> I am trying to understand if there would be cases when we still need --preservesymbols if/when the language requires that all existential/protocol types are preceded by any.

If all call sites used the existential `any` syntax, we wouldn't need to use `--preservesymbols`

Reply:
Found the quote: https://forums.swift.org/t/progress-toward-the-swift-6-language-mode/68315

> The Language Steering Group has decided that one previously-accepted upcoming feature, ExistentialAny, will not be enabled by default in Swift 6. [SE-0335: Introduce existential any](https://github.com/apple/swift-evolution/blob/main/proposals/0335-existential-any.md) introduces the any keyword to identify existential types. The proposal also specifies that "bare" protocol names will no longer be permitted as types---they must either use any or some, as appropriate---under the upcoming feature flag ExistentialAny. Given the concerns about migration to consistent use of existential any in the language, and the expectation that additional language improvements will be coming that might affect the end result of that migration, the Language Steering Group is deferring the source-incompatible changes in SE-0335 to a future language revision.

It sounds like they didn't rule out this direction, but rather just deferred it for longer (probably a very long time, if we have to wait for Swift 7!)

Reply:
In the apps repo PR that I merged today, I included `--preservesymbols canonicalFake`. This is a common and reasonable pattern that I think we should continue supporting. It's not common to use the `any ExperimentService` spelling today, so it felt too weird to require it in this specific case due to this limitation of the linter.